staging/src/k8s.io/apiserver/pkg/server/options/serving.go

```diff
@@ -260,6 +260,10 @@ func (s *SecureServingOptions) ApplyTo(config **server.SecureServingInfo) error
 			return err
 		}
 		c.CipherSuites = cipherSuites
+
+		for _, cipherName := range cliflag.InsecureTLSCipherNamesUsed(s.CipherSuites) {
```

Comment:
can you add more justification in PR description on why this check is moved here.

Reply:
The old one will check when secure server prepares to run, for example in [kube-scheduler](https://github.com/kubernetes/kubernetes/blob/master/cmd/kube-scheduler/app/server.go#L178-L185), it will check insecure cipher suites in `tlsConfig` method in `serve` function,
And the new one will check it if insecure cipher suites used when `ApplyTo` options [here](https://github.com/kubernetes/kubernetes/blob/master/cmd/kube-scheduler/app/server.go#L295-L335)  and in `config` method.
It is the same check but the new one will check it when init options rather than check it after all things are ok to run secure server. And the old one checks used cipher ids obtained by `TLSCipherSuites` function, that's not cool.

Reply:
Some components like `kube-apiserver` is as same as secure server mentioned before, its code is [here](https://github.com/kubernetes/kubernetes/blob/master/cmd/kube-apiserver/app/server.go#L454)

Reply:
This is just a warning. Not an error that will prevent server to start. So really doesn't really matter if it will be written from config Initialization or actual place where the config will be applied. I'm not very familiar with this initialization sequence. Can it happen that server will not be started because of other reasons after this warning was already written? Will it be any confusing for the customer?

One justification I can suggest is if instead of the new method `InsecureTLSCipherNamesUsed` the list of insecure ones will be returned from the `cliflag.TLSCipherSuites`. This will get rid of one more loop here.